Admin.js

```diff
@@ -114,561 +141,775 @@ var Admin = {
-        var transformedData = [];
-        for (var i = 0 ; i < data.length ; i++) {
-            transformedData[i] = {id: data[i].data, text: data[i].label};
+    "setup_sortable_select2" (subject, data) {
+
+        const transformedData = [];
+        for (let i = 0; i < data.length; i++) {
+
+            transformedData[i] = {
+                "id": data[i].data,
+                "text": data[i].label
+            };
+
         }
 
         subject.select2({
-            width: function(){
-                // Select2 v3 and v4 BC. If window.Select2 is defined, then the v3 is installed.
-                // NEXT_MAJOR: Remove Select2 v3 support.
+            "width" () {
+
+                /*
+                 * Select2 v3 and v4 BC. If window.Select2 is defined, then the v3 is installed.
+                 * NEXT_MAJOR: Remove Select2 v3 support.
+                 */
                 return Admin.get_select2_width(window.Select2 ? this.element : subject);
-            },
-            dropdownAutoWidth: true,
-            data: transformedData,
-            multiple: true
-        });
 
-        subject.select2("container").find("ul.select2-choices").sortable({
-            containment: 'parent',
-            start: function () {
-                subject.select2("onSortStart");
             },
-            update: function () {
-                subject.select2("onSortEnd");
-            }
+            "dropdownAutoWidth": true,
+            "data": transformedData,
+            "multiple": true
         });
 
+        subject.select2("container").find("ul.select2-choices").
+            sortable({
+                "containment": "parent",
+                "start" () {
+
+                    subject.select2("onSortStart");
+
+                },
+                "update" () {
+
+                    subject.select2("onSortEnd");
+
+                }
+            });
+
         // On form submit, transform value to match what is expected by server
-        subject.parents('form:first').submit(function (event) {
-            var values = subject.val().trim();
-            if (values !== '') {
-                var baseName = subject.attr('name');
-                values   = values.split(',');
-                baseName = baseName.substring(0, baseName.length-1);
-                for (var i=0; i<values.length; i++) {
-                    jQuery('<input>')
-                        .attr('type', 'hidden')
-                        .attr('name', baseName+i+']')
-                        .val(values[i])
-                        .appendTo(subject.parents('form:first'));
+        subject.parents("form:first").submit((event) => {
+
+            let values = subject.val().trim();
+            if (values !== "") {
+
+                let baseName = subject.attr("name");
+                values = values.split(",");
+                baseName = baseName.substring(0, baseName.length - 1);
+                for (let i = 0; i < values.length; i++) {
```

Comment:
- eslint: Identifier name 'i' is too short (< 2).
- eslint: Unary operator '++' used.